File: day-1/solution.c
#include <stdio.h>

/*
* Run in single line : gcc solution.c -o run & run.exe
*/
void main(){
    FILE *fp;
    int cur_value = -1;
    int prev_value = -1;
    int counter = 0;
    

    fp = fopen("input.txt", "r");
    
    fscanf(fp, "%d", &cur_value);
    prev_value = cur_value;

    while(!feof(fp)){
        fscanf(fp, "%d", &cur_value);
        if (cur_value > prev_value){
            counter ++; 
        }
        prev_value = cur_value;
    }

    fclose(fp);
    printf("Total number of increases: %d", counter);
}

File: day-1/solution.py
l = list(map(int, open('input.txt', 'r')))

prev_sum = None
counter = 0

for i in range(3, len(l) + 1):
    cur_sum = sum(l[i-3:i])
    print(l[i-3:i], prev_sum, cur_sum)

    if prev_sum is not None and cur_sum > prev_sum:
        counter += 1
    prev_sum = cur_sum

print(counter)
